aggregator/src/aggregator/taskprov_tests.rs

```diff
@@ -78,238 +78,276 @@ pub struct TaskprovTestCase {
-        },
-        vdaf_instance,
-    )
-    .with_id(task_id)
-    .with_leader_aggregator_endpoint(Url::parse("https://leader.example.com/").unwrap())
-    .with_helper_aggregator_endpoint(Url::parse("https://helper.example.com/").unwrap())
-    .with_vdaf_verify_key(vdaf_verify_key.clone())
-    .with_max_batch_query_count(max_batch_query_count as u64)
-    .with_task_expiration(Some(task_expiration))
-    .with_report_expiry_age(peer_aggregator.report_expiry_age().copied())
-    .with_min_batch_size(min_batch_size as u64)
-    .with_time_precision(Duration::from_seconds(1))
-    .with_tolerable_clock_skew(Duration::from_seconds(1))
-    .build();
-
-    let report_metadata = ReportMetadata::new(
-        random(),
-        clock
-            .now()
-            .to_batch_interval_start(task.time_precision())
-            .unwrap(),
-    );
-    let transcript = run_vdaf(
-        &vdaf,
-        vdaf_verify_key.as_ref().try_into().unwrap(),
-        &aggregation_param,
-        report_metadata.id(),
-        &measurement,
-    );
-    let report_share = generate_helper_report_share::<TestVdaf>(
-        task_id,
-        report_metadata.clone(),
-        global_hpke_key.config(),
-        &transcript.public_share,
-        Vec::new(),
-        &transcript.helper_input_share,
-    );
+        let task = TaskBuilder::new(
+            QueryType::FixedSize {
+                max_batch_size: max_batch_size as u64,
+                batch_time_window_size: None,
+            },
+            vdaf_instance,
+        )
+        .with_id(task_id)
+        .with_leader_aggregator_endpoint(Url::parse("https://leader.example.com/").unwrap())
+        .with_helper_aggregator_endpoint(Url::parse("https://helper.example.com/").unwrap())
+        .with_vdaf_verify_key(vdaf_verify_key)
+        .with_max_batch_query_count(max_batch_query_count as u64)
+        .with_task_expiration(Some(task_expiration))
+        .with_report_expiry_age(peer_aggregator.report_expiry_age().copied())
+        .with_min_batch_size(min_batch_size as u64)
+        .with_time_precision(Duration::from_seconds(1))
+        .with_tolerable_clock_skew(Duration::from_seconds(1))
+        .build();
 
-    TaskprovTestCase {
-        _ephemeral_datastore: ephemeral_datastore,
-        clock,
-        collector_hpke_keypair,
-        datastore,
-        handler: Box::new(handler),
-        peer_aggregator,
-        task,
-        task_config,
-        task_id,
-        report_metadata,
-        transcript,
-        report_share,
-        aggregation_param,
+        Self {
+            _ephemeral_datastore: ephemeral_datastore,
+            clock,
+            collector_hpke_keypair,
+            datastore,
+            handler: Box::new(handler),
+            peer_aggregator,
+            task,
+            task_config,
+            task_id,
+            vdaf,
+            global_hpke_key,
+        }
+    }
+
+    fn generate_helper_report_share(
```

Comment:
nit: this looks a lot like what `aggregator::aggregate_init_tests::PrepareInitGenerator` does. It goes and constructs a `PrepareInit`, but on the way there it does generate a VDAF transcript and generate a report share. I wonder if you could add a `PrepareInitGenerator::next_report_share_with_metadata` and use that here.

Reply:
WDYT about the latest commit? It's not really a simplification IMO, but I like that it expands on PrepareInitGenerator to make other kinds of tests easier to write.